### PR TITLE
Save reqExec in Architecture/OS dependent path

### PR DESCRIPTION
**Pull request by shrivara**
https://github.com/Shippable/reqProc/issues/182

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,19 +1,20 @@
 FROM {{%DRYDOCK_ORG%}}/microbase:{{%TAG%}}
 
-ENV REQPROC_PATH /home/shippable/reqProc
-ADD . $REQPROC_PATH
-RUN cd $REQPROC_PATH && npm install
+# Prefix all image ENVs with IMAGE_ so there are no confusions.
```
Comment (shrivara) [inline Dockerfile]: Added this convention as the env names could get confusing as we go along.

File: app.js [hunk 2]
```diff
@@ -41,9 +41,6 @@ if (!global.config.reqProcDir)
 if (!global.config.reqExecDir)
   consoleErrors.push(util.format('%s is missing: reqExecDir', who));
 
-if (!global.config.reqExecBinDir)
```
Comment (shrivara) [inline app.js]: This is not used anymore. It comes from an ENV.

File: Dockerfile [hunk 3]
```diff
@@ -1,19 +1,20 @@
...
     wget https://github.com/Shippable/execTemplates/archive/{{%TAG%}}.tar.gz -O /tmp/execTemplates.tar.gz && \
-    tar -xzvf /tmp/execTemplates.tar.gz -C $EXEC_TEMPLATES_PATH --strip-components=1 && \
+    tar -xzvf /tmp/execTemplates.tar.gz -C $IMAGE_EXEC_TEMPLATES_DIR --strip-components=1 && \
     rm /tmp/execTemplates.tar.gz
 
-ENV REQEXEC_PATH /home/shippable/reqExec
-RUN mkdir -p $REQEXEC_PATH && \
-    wget https://s3.amazonaws.com/shippable-artifacts/reqExec/{{%TAG%}}/reqExec-{{%TAG%}}-{{%ARCHITECTURE%}}-{{%OS%}}.tar.gz -O /tmp/reqExec.tar.gz && \
-    tar -xzvf /tmp/reqExec.tar.gz -C $REQEXEC_PATH && \
+ENV IMAGE_REQEXEC_DIR /home/shippable/reqExec
```
Comment (shrivara) [inline Dockerfile]: We are pulling `IMAGE_REQEXEC_DIR/x86_64/Ubuntu_16.04`. The rest of them should also be explicitly pulled once we have them in S3.

File: Dockerfile [hunk 4]
```diff
@@ -1,19 +1,20 @@
 FROM {{%DRYDOCK_ORG%}}/microbase:{{%TAG%}}
 
-ENV REQPROC_PATH /home/shippable/reqProc
-ADD . $REQPROC_PATH
-RUN cd $REQPROC_PATH && npm install
+# Prefix all image ENVs with IMAGE_ so there are no confusions.
+ENV IMAGE_REQPROC_DIR /home/shippable/reqProc
```
Comment (ric03uec) [inline Dockerfile]: we should get rid of /home/shipapble here. it should be /root 
- all new workflow artifacts should be stored in /root

File: Dockerfile (hunk 3 above)
Comment (ric03uec) [inline Dockerfile]: chagne dir to /root

**Pull request merged**